include/jemalloc/internal/background_thread_structs.h

```diff
@@ -11,6 +11,9 @@
 #define MAX_BACKGROUND_THREAD_LIMIT MALLOCX_ARENA_LIMIT
 #define DEFAULT_NUM_BACKGROUND_THREAD 4
 
+#define BACKGROUND_THREAD_HPA_INTERVAL_MAX_UNINITIALIZED (-2)
+#define BACKGROUND_THREAD_HPA_INTERVAL_MAX_DEFAULT_WHEN_ENABLED 5000
```

Comment:
Honestly, it's mostly about convenience for now. Eventually, the PAI should be expanded so that the implementations can pass back information about when they've deferred work to.

(Knowing that this was coming was one of my ulterior motivations for the background thread stuff as a starter project -- "improve the PAI interface to take deferral" is a good segue from learning about the background thread code into learning about the allocator backends).

Reply:
IIUC, so long as we are able to return the deferred work info from the PAI, the max interval is not longer necessary? Right now it's just there to make sure that the deferred actions will happen.

So the downside of not having it, is that if things go idle, we may never hugify the eligible candidates?

Reply:
> IIUC, so long as we are able to return the deferred work info from the PAI, the max interval is not longer necessary? Right now it's just there to make sure that the deferred actions will happen.

Right; the interface I'm thinking of is that the PAI methods should take an `nstime_t *r_max_deferral_interval` or something, and a new method to do the deferred work. Once we've done that, we don't need a max sleep interval any more.

> So the downside of not having it, is that if things go idle, we may never hugify the eligible candidates?

Yeah (well, or purge them).

Reply:
Sounds good. Could you also comment a bit on the `max_interval`, saying that it's only temporary?

Reply:
Done.